### DISCUSSION TITLE
Android support

### DISCUSSION THREAD

**Discussion by stephomi** [elalish/manifold#514] [Q&A]
Few minor issues that I encountered integrating manifold on Android

- hard crash on armv7, not during the Boolean operation but simply when launching my app on the device. I suspect a thrust thing. I won’t investigate further as armv7 is old crap but I’m curious to know if someone else managed to make it work on armv7 (manifold works on wasm so I see no reason why it wouldn’t work). For now I simply don’t use manifold for armv7.
- same as issue as Apple, my NDK 25 doesn’t seem to support some cpp features https://github.com/stephomi/manifold/commit/222890c7ce2f87b1d16d3f5b14dcd2b626f83c62. I haven’t checked ndk beta 26 but hopefully support will be there, so waiting a bit is fine for me (q3 2023)

**Comment by elalish** [comment]
You're the first I know of trying it on Android, but that's great! If you know a good way to test it in our CI, feel free to add a build to make sure we don't break. Regarding your change, I just merged something to replace `__APPLE__` checks that should be more robust, so hopefully you won't need your tweak anymore.

**Comment by stephomi** [comment]
Yep the new preprocessor check works fine!
Very minor thing but the README https://github.com/elalish/manifold#manifold is missing the `quickhull` dependency

I don't think it's worth the hassle to test mobile builds (pulling the whole NDK, etc). I don't know much about github CI though.
Typically for iOS, `brew install tbb` pulls a library for macos, not iOS. So I built separately TBB for iOS.
For Android I am using OMP for now. It should be possible to use TBB but I couldn't make it work easily (Android is less important to me so I quickly gave up).

**Comment by pca006132** [comment]
It seems that nix supports cross-compiling to Android platform, and if the binary cache works correctly it should not take forever on the CI to build things.
Testing might require running QEMU though.

**Comment by elalish** [comment]
Beware that we have found OMP not to actually help with performance - there's even been talk of removing it since thrust doesn't seem to support it well. Let us know if you find otherwise.
